### ISSUE TITLE
Add .NET8 support and resolve Microsoft.Bcl.AsyncInterfaces conflicts

### ISSUE THREAD

**Issue by mr-russ** [stripe/stripe-dotnet#2812]
### Is your feature request related to a problem? Please describe.

Yes.  I've upgraded my libraries and application to .NET8, and Framework 4.8  Other libraries depend on;
assembly ```'Microsoft.Bcl.AsyncInterfaces, Version=8.0.0.0'```

But stripe depends on; ```'Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0'```

I've been unable to resolve this conflict as I cannot upgrade other libraries as stripe will not remap to version 8.  And I think it's trying to be a .netstandard target rather than framework 4.8.  Which the Framework projects are.

We have begun having more difficulty with .NETStandard and when upgrading to .NET8, some issues were not able to be resolved.  So we have moved to multi-target solution rather than libraries that use .NETStandard.

### Describe the solution you'd like

Add net8.0 and net48 targets to the build and depend on Microsoft.Bcl.AsyncInterfaces version 8.0.0.0 rather than earlier versions.  Or not version specifically.

### Describe alternatives you've considered

1. Compiling my own code
2. Not using the .NET stripe library

Not upgrading is not really an issue as we need to keep security support.

**Comment by pakrym-stripe** [comment]
Can you share a csproj that reproduces the issue?

I was not able to reproduce with the following:
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net8.0;net4.8</TargetFrameworks>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <LangVersion>latest</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
    <PackageReference Include="Stripe.net" Version="43.6.0" />
  </ItemGroup>
</Project>
```

**Comment by mr-russ** [comment]
I will make a version and see if I can make a minimal reproduction.  The issue does not occur until runtime.  So it will build fine and say there aren't any problems.

With the minimal project you need to make a stripe API call.

**Comment by mr-russ** [comment]
See: https://github.com/stripe/stripe-dotnet/compare/master...mr-russ:stripe-dotnet:master

That version will fail the: StripeTests.AllStripeObjectClassesPresentInDictionary.Check

With that version. If you remove the Npgsql package, it will pass.  Put it back, it will fail.

It's not the exact issue I was experiencing in my code.  But I think they are related and likely have the same root cause.

**Comment by richardm-stripe** [comment]
Hello @mr-russ. Do you think you could provide an even more minimal reproduction? The link you provided appears to reference `..\..\..\..\..\..\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.8\System.Net.Http.dll` and require windows and/or your specific system to run.

Ideally a minimal repro would be an independent project that anybody could `git clone` from the Internet and get running with just a basic .NET environment and the Stripe.Net dependency.

It also would be good to see the exact text representing the error/unexpected behavior you are seeing. On the face of it `StripeTests.AllStripeObjectClassesPresentInDictionary.Check` failing doesn't really seem connected to the version of `Microsoft.Bcl.AsyncInterfaces`

**Comment by mr-russ** [comment]
I'm still trying to build smaller test cases that reproduce the error.  It's not simple.  I've spent over a day on it so far.  the case given at least produces errors about incompatibility.

After turning on assembly loading failure debugging, I have the following from my original project

```
 Stack Trace: 
   CardChargeTests.TestChargeInvoice() line 154
   1)    at WebApp.Tests.CardChargeTests.TestChargeInvoice() in C:\...\WebApp.Tests\CardChargeTests.cs:line 154

 Standard Output: 
   2023-12-13 15:08:12.7860 INFO 16 Website.CardChargeController.ChargeInvoices(CardCharge.aspx.cs:131) System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   File name: 'Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' ---> System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   File name: 'Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
   
   === Pre-bind state information ===
   LOG: DisplayName = Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
    (Fully-specified)
   LOG: Appbase = file:///C:/.../WebApp.Tests/bin/Debug/
   LOG: Initial PrivatePath = NULL
   Calling assembly : Stripe.net, Version=43.6.0.0, Culture=neutral, PublicKeyToken=null.
   ===
   LOG: This bind starts in default load context.
   LOG: Using application configuration file: C:\...\WebApp.Tests\bin\Debug\WebApp.Tests.dll.config
   LOG: Using host configuration file: 
   LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config.
   LOG: Redirect found in application configuration file: 1.0.0.0 redirected to 6.0.0.0.
   LOG: Post-policy reference: Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
   LOG: Attempting download of new URL file:///C:/.../WebApp.Tests/bin/Debug/Microsoft.Bcl.AsyncInterfaces.DLL.
   WRN: Comparing the assembly name resulted in the mismatch: Major Version
   ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.
   
      at Backend.Utilities.StripeController.StripeCharge(String currency, String amountInCents, String stripeCustomerId, String description)
      at Website.CardChargeController.ChargeInvoices(JObject data) in C:\...\CardCharge.aspx.cs:line 131
```

**Comment by mr-russ** [comment]
I have been unable to reproduce the issue when making a simpler test case.  There are dependency issues on our project somewhere and it was only showing up here.  The project I posted is a strange error when a new library is added to the regular stripe project.

After a few more hours attempting to reproduce the issue, I managed to resolve it in our project and haven't been able to make a reasonable test case.

So closing this as I can't reproduced it.  Thanks for your time and responses.
